dataset_loaders/images/mnist.py

```diff
@@ -0,0 +1,186 @@
+"""Mnist dataset."""
+import os
+import time
+
+import numpy as np
+from PIL import Image
+
+from dataset_loaders.parallel_loader import ThreadedDataset
+
+
+class MnistDataset(ThreadedDataset):
+    """The mnist handwritten digit dataset.
+
+    The dataset should be downloaded from [MNIST] into the `shared_path`
+    (that should be specified in the config.ini according to the
+    instructions in ../README.md).
+
+    Parameters
+    ----------
+    which_set: string
+        A string in ['train', 'val', 'valid', 'test'], corresponding to
+        the set to be returned.
+
+     References
+    ----------
+    [MNIST] Mnist dataset:
+        http://yann.lecun.com/exdb/mnist/
+    """
+
+    name = 'mnist'
+
+    # optional arguments
+    data_shape = (28, 28, 3)
```

Comment:
I believe MNIST data_shape should be `(28, 28)`. Can you verify this?

Reply:
It is in some cases. I can change this if you want. Also, mnistm has 3 channels. Can it still be inherited from mnist?

Reply:
No problem :)

Reply:
:)